src/reducers/selectors.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import { createSelector } from 'reselect';
+
+export const EMPTY_OBJ = {};
+export const EMPTY_AR = [];
+
+// Checks a response object for errors and returns them
+export const processErrors = (responseValue: ?QueryResponse): Object => {
+	if (!responseValue || !responseValue.error) {
+		return {};
+	}
+	return responseValue;
+};
+
+export const processErrorsMaybe = (responseValue: ?QueryResponse) => {
+	const processed = processErrors(responseValue);
+	return Object.keys(processed).length ? processed : null;
+};
+
+// Checks if object is empty
+export const isEmpty: Selector<boolean> = state => {
+	if (!state) {
+		return true;
+	}
+	return Object.keys(state).length === 0;
+};
+
+// Checks if object is has errors
+export const hasErrors: Selector<boolean> = state =>
+	Object.keys(processErrors(state)).length > 0;
+
+// Combo check if the object is either empty or has errors
+export const hasValidValue: Selector<boolean> = state =>
+	!isEmpty(state) && !hasErrors(state);
+
+/*
+ * returns the QueryResponse.value, or an object containing a single `error`
+ * property. An empty response returns the defaultValue.
+ */
+export const getValue = (
+	resp: ?QueryResponse,
+	defaultValue: mixed = EMPTY_OBJ
+): mixed => {
+	if (!resp || isEmpty(resp)) {
```

Comment:
Why do we need the extra `isEmpty` check? If the returned value is an array, it will return an empty object if `defaultValue` is not set. Seems like it could lead to weird bugs?